projects/common/Utils/DebugHandler.h

```diff
@@ -37,82 +40,81 @@ enum PROGRAM_TYPE
 };
 
 #define NC_LOG_MESSAGE(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::Print(message, __VA_ARGS__);
+DebugHandler::Print(message);
 
 #define NC_LOG_WARNING(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::PrintWarning(message, __VA_ARGS__);
+DebugHandler::PrintWarning(message);
 
 #define NC_LOG_DEPRECATED(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::PrintDeprecated(message, __VA_ARGS__);
+DebugHandler::PrintDeprecated(message);
 
 #define NC_LOG_ERROR(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::PrintError(message, __VA_ARGS__);
+DebugHandler::PrintError(message);
 
 #define NC_LOG_FATAL(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::PrintFatal(message, __VA_ARGS__);
+DebugHandler::PrintFatal(message);
 
 #define NC_LOG_SUCCESS(message, ...) if (!DebugHandler::isInitialized) { DebugHandler::Initialize(); } \
-DebugHandler::PrintSuccess(message, __VA_ARGS__);
+DebugHandler::PrintSuccess(message);
 
 class DebugHandler
 {
 public:
     static bool isInitialized;
     static void Initialize();
 
-    template <typename... Args>
-    inline static void Print(std::string message, Args... args)
```

Comment:
Clang complained about this too, so I just removed it. As per the previous comment we might want to think about reworking this so that it is cross-platform and cross-compiler by default.

Reply:
Yes, I think the issue is that Clang doesn't like us not passing any args. So the whole "you can pass any amount of arguments here" becomes "you can pass any amount of arguments here, except 0" for some incredibly stupid reason. Maybe it's enough to copy it and change the copys template line to "template <>". I can't really look into this more detailed than this at the moment. :/